File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F15.P3.go
package main

import "fmt"

// BEGIN PROBE F15.P3
type Named interface{ tag() string }

type A struct{}

func (A) tag() string { return "a" }

type B struct{}

func (B) tag() string { return "b" }

func firstTag() string {
	items := []Named{A{}, B{}}
	return items[0].tag()
}

// END PROBE F15.P3

func main() {
	fmt.Println(firstTag())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F15.P3.ts
// BEGIN PROBE F15.P3
interface Named {
tag(): string
}

class A implements Named {
tag(): string {
return "a"
}
}

class B implements Named {
tag(): string {
return "b"
}
}

function firstTag(): string {
const items: Named[] = [new A(), new B()]
return items[0].tag()
}
// END PROBE F15.P3

console.log(firstTag())
